dvc/remote/gs.py

```diff
@@ -65,11 +65,82 @@ def _upload_to_bucket(
             blob.upload_from_file(wrapped)
 
 
+class GSRemoteTree(BaseRemoteTree):
+    @property
+    def gs(self):
+        return self.remote.gs
```

Comment:
This is kinda strange. Shouldn't the connection be part of the Tree?

Reply:
Or is it just a convenience that will be fixed after the transition?

Reply:
I think all of the connection/config related code for all of the remotes needs to be moved into the trees. This (and the equivalent properties in the other remotes) is just a temporary fix since I was trying to avoid moving everything all at once